noodles-editor/src/noodles/noodles.tsx

```diff
@@ -615,6 +618,11 @@ export function getNoodles(): Visualization {
     }))
   }, [edges])
 
+  const handleProjectUpdate = useCallback((updatedProject: { nodes: AnyNodeJSON[], edges: ReactFlowEdge<unknown>[] }) => {
```

Comment:
Does this update all of the state? What about `opMap`?

Reply:
#43 clarified this.. everything is driven by node and edge changes. opMap is updated by transformGraph